src/javascript/JContent/actions/createFolderAction/createFolderAction.jsx

```diff
@@ -1,21 +1,54 @@
-import React from 'react';
-import {composeActions, componentRendererAction} from '@jahia/react-material';
-import requirementsAction from '../requirementsAction';
+import React, {useContext} from 'react';
 import CreateFolderDialog from './CreateFolderDialog';
+import {useNodeChecks} from '@jahia/data-helper';
+import {ComponentRendererContext} from '@jahia/ui-extender';
+import PropTypes from 'prop-types';
 
-export default composeActions(requirementsAction, componentRendererAction, {
-    init: context => {
-        context.initRequirements({requiredPermission: 'jcr:addChildNodes'});
-    },
-    onClick: context => {
-        const onExit = () => {
-            handler.destroy();
-        };
-
-        let handler = context.renderComponent(
-            <CreateFolderDialog node={context.node}
-                                contentType={context.contentType}
-                                onExit={onExit}/>
-        );
+export const CreateFolderAction = ({context, render: Render, loading: Loading}) => {
+    const componentRenderer = useContext(ComponentRendererContext);
+    const res = useNodeChecks(
+        {path: context.path},
+        {requiredPermission: ['jcr:addChildNodes'], ...context}
```

Comment:
what is ...context for ?

Reply:
because the showOnNodeTypes coming from configuration is not applied without adding it here

Reply:
ok